src/configure/helper/repoAnalysisHelper.ts

```diff
@@ -46,32 +47,35 @@ export class RepoAnalysisHelper {
                 let applicationSettings: RepositoryAnalysisApplicationSettings = new RepositoryAnalysisApplicationSettings();
                 applicationSettings.language = analysis.language;
 
-                if(!!analysis.settings && !!analysis.buildTargetName) {
-                    applicationSettings.buildTargetName = analysis.buildTargetName;
-                    if (analysis.language === SupportedLanguage.NODE) {
-                        applicationSettings.settings.nodePackageFilePath = analysis.settings[RepoAnalysisConstants.PackageFilePath];
-                        applicationSettings.settings.nodePackageFileDirectory = path.dirname(analysis.settings[RepoAnalysisConstants.PackageFilePath]);
-                        if (analysis.buildTargetName === RepoAnalysisConstants.Gulp && !!analysis.settings[RepoAnalysisConstants.GulpFilePath]) {
-                            applicationSettings.settings.nodeGulpFilePath = analysis.settings[RepoAnalysisConstants.GulpFilePath];
+                if(!!analysis.settings){
+                    applicationSettings.settings.workingDirectory = !!analysis.settings.workingDirectory ? analysis.settings.workingDirectory : workspacePath;
```

Comment:
Why do we need this variable called `applicationSettings`. can't we directly use `analysis` object?

Reply:
Analysis object can be changed if service chose to update their model but applicationSetting will be same which is known by rest of code in our extension.